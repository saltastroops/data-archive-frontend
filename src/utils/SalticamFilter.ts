export type SalticamFilter =
  | "SalticamFilter_Name"
  | "340-35"
  | "380-40"
  | "B-S1"
  | "CLR-S1"
  | "Halpha-S1"
  | "Hbn-S1"
  | "Hbw-S1"
  | "I-S1"
  | "R-S1"
  | "Sb-S1"
  | "SDSSg-S1"
  | "SDSSi-S1"
  | "SDSSr-S1"
  | "SDSSu-S1"
  | "SDSSz-S1"
  | "SR613-21"
  | "SR708-25"
  | "SR815-29"
  | "SR862-32"
  | "Su-S1"
  | "Sv-S1"
  | "Sy-S1"
  | "U-S1"
  | "V-S1";

export interface ISalticamFilter {
  name: SalticamFilter;
  descriptiveName: string;
}
