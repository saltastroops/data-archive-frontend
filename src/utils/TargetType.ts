export type TargetType = string;

export const GALAXY = "10";

export const ISM = "20";

export const SOLAR_SYSTEM_BODY = "30";

export const STAR = "40";
