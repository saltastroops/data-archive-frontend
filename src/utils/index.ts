export const toTitleCase = (s: string) => {
  s = s.replace(/_/g, " ");
  return s.replace(/\w\S*/g, (txt: string) => {
    return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
  });
};
