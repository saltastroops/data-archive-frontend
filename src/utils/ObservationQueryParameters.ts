// TypeScript interfaces describing the state of the search form

import { SalticamFilter } from "./SalticamFilter";
import { TargetType } from "./TargetType";

/**
 * An interface describing the React state of the observation query form.
 *
 * This state is used for populating all search form fields (and error messages)
 * and is parsed into the GraphQL query for searching observations.
 *
 * Properties:
 * -----------
 * general:
 *     General information about the observation.
 * target:
 *     Target details.
 * telescope:
 *     Telescope (and instrument) details.
 */
// TODO: Remove loading, replace with IObservationQueryParameters
export interface ISearchFormState {
  general: IGeneral;
  target: ITarget;
  telescope?: ITelescope;
  loading: boolean;
}

/**
 * The available calibration types.
 */
export type CalibrationType = "arc" | "bias" | "flat" | "standard";

/**
 * An interface for query parameters related to general information.
 *
 * Properties:
 * -----------
 * errors:
 *     Error messages.
 * observationNight:
 *     Date of the night when the observation was taken. This
 *     refers to the date whe the night started.
 * proposalCode:
 *     Identifier of the proposal to which the observation belongs.
 * proposalTitle:
 *     Title of the proposal to which the observation belongs.
 * calibrations:
 *     Calibration types to include in the search results
 *
 */
export interface IGeneralErrors {
  observationNight?: string;
  principalInvestigator?: string;
  proposalCode?: string;
  proposalTitle?: string;
  calibrations?: string;
}
export interface IGeneral {
  errors: IGeneralErrors;
  observationNight?: string;
  principalInvestigator?: string;
  proposalCode?: string;
  proposalTitle?: string;
  calibrations: Set<CalibrationType>;
}

/**
 * The available target name resolvers.
 */
export type TargetResolver = "Simbad" | "NED" | "VizieR";

/**
 * The available units for the search cone radius.
 */
export type SearchConeRadiusUnits = "arcseconds" | "arcminutes" | "degrees";

/**
 * An interface for query parameters related to target details.
 *
 * Properties:
 * -----------
 * declination:
 *     Declination.
 * errors:
 *     Errors.
 * name:
 *     Target name.
 * resolver:
 *     Resolver to use for resolving the target name to a position.
 * rightAscension:
 *     Right ascension.
 * searchConeRadius:
 *     Radius of the search cone radius, in the units given by
 *     searchConeRadiusUnits.
 * searchConeRadiusUnits:
 *     Units of the search cone radius given by searchConeRadius.
 *
 */
export interface ITarget {
  declination?: string;
  errors: ITargetErrors;
  loading?: string;
  name?: string;
  resolver: TargetResolver;
  rightAscension?: string;
  searchConeRadius?: string;
  searchConeRadiusUnits: SearchConeRadiusUnits;
  targetTypes: Set<TargetType>;
}

export interface ITargetErrors {
  name?: string;
  declination?: string;
  resolver?: string;
  rightAscension?: string;
  searchConeRadius?: string;
  searchConeRadiusUnits?: string;
  targetTypes?: string;
}

// TELESCOPES

/**
 * The supported telescope names.
 */
export type TelescopeName = "Lesedi" | "SALT" | "1.9 m";

/**
 * An interface for query parameters related to a telescope.
 *
 * This serves as a placeholder for more concrete interfaces such as ISALT.
 *
 * Properties:
 * -----------
 * name:
 *     Telescope name.
 */
export interface ITelescope {
  name?: TelescopeName;
}

/**
 * An interface for query parameters related to SALT.
 *
 * Properties:
 * -----------
 * instrument:
 *     A SALT instrument.
 * name:
 *     The string "SALT".
 */
export interface ISALT extends ITelescope {
  instrument: IInstrument;
  name: "SALT";
}

/**
 * An interface for query parameters related to Lesedi.
 *
 * Properties:
 * -----------
 * instrument:
 *     An instrument used on Lesedi.
 * name:
 *     The string "Lesedi".
 */
export interface ILesedi extends ITelescope {
  instrument: IInstrument;
  name: "Lesedi";
}

/**
 * An interface for query parameters related to the 1.9 m Telescope.
 *
 * Properties:
 * -----------
 * instrument:
 *     An instrument used on the 1./9 m Telescope.
 * name:
 *     The string "1.9 m".
 */
export interface IOneDotNineM extends ITelescope {
  instrument: IInstrument;
  name: "1.9 m";
}

// INSTRUMENTS

/**
 * The available instrument names.
 */
export type InstrumentName =
  | "Salticam"
  | "RSS"
  | "HRS"
  | "BVIT"
  | "SpUpNIC"
  | "SHOC"
  | "HIPPO";

/**
 * An interface for query parameters related to an instrument.
 *
 * This serves as a placeholder for more concrete interfaces such as ISalticam.
 *
 * Properties:
 * -----------
 * name:
 *     Telescope name.
 */
export interface IInstrument {
  name: InstrumentName;
}

/**
 * An interface for query parameters related to Salticam.
 *
 * Properties:
 * -----------
 * detectorMode:
 *     Detector mode.
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "Salticam".
 */
export interface ISalticam extends IInstrument {
  detectorMode?: "Normal" | "Slot Mode";
  errors: {
    detectorMode?: string;
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: SalticamFilter;
  name: "Salticam";
}

/**
 * An interface for query parameters related to RSS.
 *
 * Properties:
 * -----------
 * detectorMode:
 *     Detector mode.
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * mode:
 *     Instrument mode specific details.
 * name:
 *     The string "RSS".
 */
export interface IRSS extends IInstrument {
  detectorMode?: RSSDetectorMode;
  errors: {
    detectorMode?: string;
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  modes?: IRSSModes;
  name: "RSS";
}

export interface IRSSModes {
  errors: {
    fabryPerotMode?: string;
    grating?: string;
    polarimetryModes?: string;
  };
  fabryPerotMode?: RSSFabryPerotMode;
  grating?: RSSGrating;
  names: Set<RSSInstrumentMode>;
  polarimetryModes?: Set<RSSPolarimetryMode>;
}

export type RSSDetectorMode = "Normal" | "Slot Mode" | "";

export type RSSGrating =
  | "Open"
  | "pg0300"
  | "pg0900"
  | "pg1300"
  | "pg1800"
  | "pg2300"
  | "pg3000"
  | "";

export type RSSInstrumentMode =
  | "Fabry Perot"
  | "FP polarimetry"
  | "Imaging"
  | "Polarimetric imaging"
  | "MOS"
  | "MOS polarimetry"
  | "Spectropolarimetry"
  | "Spectroscopy";

export type RSSFabryPerotMode = "HR" | "LR" | "MR" | "TF" | "";

export type RSSPolarimetryMode =
  | "All Stokes"
  | "Circular"
  | "Linear"
  | "Linear Hi"
  | "";

/**
 * An interface for query parameters related to HRS.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * mode:
 *     Resolution mode.
 * name:
 *     The string "HRS".
 */
export interface IHRS extends IInstrument {
  errors: {
    exposureTime?: string;
    mode?: string;
  };
  exposureTime?: string;
  mode?: HRSMode;
  name: "HRS";
}

export type HRSMode =
  | "High Resolution"
  | "High Stability"
  | "Int Cal Fibre"
  | "Low Resolution"
  | "Medium Resolution";

export interface IBVIT extends IInstrument {
  errors: {};
  filter?: BVITFilter;
  mode?: "Imaging" | "Streaming";
  name: "BVIT";
}

export type BVITFilter = "B" | "H-alpha" | "Open" | "R" | "U" | "V";

/**
 * An interface for query parameters related to HIPPO.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "HIPPO".
 */
export interface IHIPPO extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "HIPPO";
}

/**
 * An interface for query parameters related to SHOC.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "SHOC".
 */
export interface ISHOC extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "SHOC";
}

/**
 * An interface for query parameters related to SpUpNIC.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "SpUpNIC".
 */
export interface ISpUpNIC extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "SpUpNIC";
}

export interface IFile {
  id: string;
  filename: string;
  name: string;
  dataType: string;
  isReduced: boolean;
  targetName: string;
  rightAscension: number;
  declination: number;
  observationNight: string;
  category: string;
  telescope: string;
  instrument: string;
  observationId?: string;
  url?: string;
}
export interface IObservation {
  id: string;
  name: string | "Unknown Observation";
  proposal?: string;
  telescope: string;
  startTime: string;
  files: IFile[];
}
