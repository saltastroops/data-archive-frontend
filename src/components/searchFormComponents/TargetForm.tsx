import * as React from "react";
import targetPosition from "target-position";
import { ITarget } from "../../utils/ObservationQueryParameters";
import { TargetType } from "../../utils/TargetType";
import {
  validateDeclination,
  validateRightAscension,
  validateSearchConeRadius
} from "../../utils/validators";
import {
  InnerMainGrid,
  MainGrid,
  SingleColumnGrid,
  SubGrid
} from "../basicComponents/Grids";
import InputField from "../basicComponents/InputField";
import SelectField from "../basicComponents/SelectField";
import TargetTypesSelector from "./TargetTypesSelector";

interface ITargetFormProps {
  target: ITarget;
  onChange: (value: ITarget) => any;
}

interface ITargetFormState {
  loading: boolean;
}

/**
 * A form for providing target-related search parameters for an observation
 * search.
 */
class TargetForm extends React.Component<ITargetFormProps, ITargetFormState> {
  state = { loading: false };

  // Function for updating the loading property of the state
  updateLoadingStatus = (loading: boolean) => {
    this.setState(() => ({
      loading
    }));
  };

  // Function for resolving the target name to a position
  resolve = async () => {
    const { target, onChange } = this.props;
    this.updateLoadingStatus(true);
    const resolver = target.resolver;
    onChange({
      ...target
    });

    try {
      const p = await targetPosition(target.name || "", [resolver]);
      if (p) {
        onChange({
          ...target,
          declination: `${p.declination}`,
          rightAscension: `${p.rightAscension}`
        });
      } else {
        onChange({
          ...target,
          declination: ``,
          errors: {
            ...target.errors,
            name: `Target ${target.name} could not be resolved.`
          },
          rightAscension: ``
        });
      }
    } catch (err) {
      onChange({
        ...target,
        errors: {
          ...target.errors,
          name: err.message
        }
      });
    }
    this.updateLoadingStatus(false);
  };

  render() {
    const { target, onChange } = this.props;
    const { loading } = this.state;

    // Function for handling changes made to the search parameters
    const handleChangeEvent = (
      e: React.ChangeEvent<HTMLSelectElement | HTMLInputElement>
    ) => {
      const name = e.target.name;
      const value = e.target.value || "";
      targetChange(name, value);
    };

    // Function for handling target type changes
    const handleTargetTypeChange = (targetTypes: Set<TargetType>) => {
      targetChange("targetTypes", targetTypes);
    };

    const targetChange = (property: string, value: any) => {
      onChange({
        ...target,
        [property]: value,
        errors: {
          ...target.errors,
          [property]: ""
        }
      });
    };

    return (
      <>
        <MainGrid>
          <SubGrid>
            <p>Target name</p>
            <InputField
              data-test="target-name-input"
              disabled={loading}
              name={"name"}
              value={target.name || ""}
              error={target.errors.name}
              onChange={handleChangeEvent}
            />
          </SubGrid>
          <SubGrid>
            <InnerMainGrid>
              <SubGrid>
                <p>Resolver</p>
                <SelectField
                  data-test={"resolver-select"}
                  name={"resolver"}
                  value={target.resolver}
                  onChange={handleChangeEvent}
                >
                  <option value="Simbad">Simbad</option>
                  <option value="NED">NED</option>
                  <option value="VizieR">VizieR</option>
                </SelectField>
              </SubGrid>
              <SubGrid>
                <br />
                <button
                  className={`button is-info ${loading &&
                    "is-loading disable"}`}
                  type="button"
                  onClick={this.resolve}
                  title={`${loading ? "Loading." : ""}`}
                  disabled={loading}
                >
                  resolve
                </button>
              </SubGrid>
            </InnerMainGrid>
          </SubGrid>
        </MainGrid>

        <MainGrid>
          <SubGrid>
            <p>Right ascension</p>
            <InputField
              data-test="right-ascension-input"
              disabled={loading}
              name={"rightAscension"}
              value={target.rightAscension || ""}
              onChange={handleChangeEvent}
              error={target.errors.rightAscension}
            />
          </SubGrid>
          <SubGrid>
            <p>Declination</p>
            <InputField
              data-test="declination-input"
              disabled={loading}
              name={"declination"}
              value={target.declination || ""}
              onChange={handleChangeEvent}
              error={target.errors.declination}
            />
          </SubGrid>
        </MainGrid>
        <MainGrid>
          <SubGrid>
            <p>Search radius</p>
            <InputField
              data-test="search-cone-radius-input"
              name={"searchConeRadius"}
              value={target.searchConeRadius || ""}
              onChange={handleChangeEvent}
              error={target.errors.searchConeRadius}
            />
          </SubGrid>
          <SubGrid>
            <p>Radius units</p>
            <SelectField
              data-test="radius-units-select"
              name={"radiusUnits"}
              onChange={handleChangeEvent}
              value={target.searchConeRadiusUnits}
            >
              <option value="arcseconds">Arcseconds</option>
              <option value="arcminutes">Arcminutes</option>
              <option value="degrees">Degrees</option>
            </SelectField>
          </SubGrid>
        </MainGrid>

        <SingleColumnGrid>
          <SubGrid>
            <p>Target type</p>
            <TargetTypesSelector
              onChange={handleTargetTypeChange}
              targetTypes={target.targetTypes}
            />
          </SubGrid>
        </SingleColumnGrid>
      </>
    );
  }
}

/**
 * Validate the given target-related search parameters and, if need be, add
 * error messages to them.
 */
export const validatedTarget = async (target: ITarget) => {
  // Check that the target name and any given coordinates are consistent
  let raDecChangeError;
  if (target.name && target.name !== "") {
    raDecChangeError = await targetPosition(target.name, [target.resolver])
      .then(position => {
        if (position) {
          return {
            declination:
              `${position.declination}` !== target.declination
                ? `The given target name resolves to a different declination with ${
                    target.resolver
                  }.`
                : "",
            rightAscension:
              `${position.rightAscension}` !== target.rightAscension
                ? `The given target name resolves to a different right ascension with ${
                    target.resolver
                  }.`
                : ""
          };
        }
      })
      .catch();
  }

  let targetNameError = "";
  if (target.name) {
    targetNameError = (await targetPosition(target.name, [target.resolver]))
      ? ""
      : `Target "${target.name}" Could not be resolved.`;
  }
  // Return the search parameters with the errors found
  return {
    ...target,
    errors: {
      ...target.errors,
      declination:
        raDecChangeError && raDecChangeError.declination
          ? raDecChangeError.declination
          : validateDeclination(target.declination || ""),
      name: targetNameError,
      rightAscension:
        raDecChangeError && raDecChangeError.rightAscension
          ? raDecChangeError.rightAscension
          : validateRightAscension(target.rightAscension || ""),
      searchConeRadius: validateSearchConeRadius(target.searchConeRadius || "")
    }
  };
};

export default TargetForm;
