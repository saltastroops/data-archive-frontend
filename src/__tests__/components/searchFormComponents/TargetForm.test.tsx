import { mount, ReactWrapper, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import TargetForm from "../../../components/searchFormComponents/TargetForm";
import { TargetType } from "../../../utils/TargetType";

describe("TargetForm", () => {
  const onChange = jest.fn();

  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(
      <TargetForm
        target={{
          errors: {},
          resolver: "Simbad",
          searchConeRadiusUnits: "arcseconds",
          targetTypes: new Set<TargetType>()
        }}
        onChange={onChange}
      />
    );

    onChange.mockReset();
  });

  it("should render", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(
      toJson(
        shallow(
          <TargetForm
            target={{
              errors: {},
              resolver: "Simbad",
              searchConeRadiusUnits: "arcseconds",
              targetTypes: new Set<TargetType>()
            }}
            onChange={onChange}
          />
        )
      )
    ).toMatchSnapshot();

    expect(
      toJson(
        shallow(
          <TargetForm
            target={{
              errors: {},
              name: "NGC 1234",
              resolver: "Simbad",
              searchConeRadiusUnits: "arcseconds",
              targetTypes: new Set<TargetType>()
            }}
            onChange={onChange}
          />
        )
      )
    ).toMatchSnapshot();

    expect(
      toJson(
        shallow(
          <TargetForm
            target={{
              errors: {
                name: "this is an error"
              },
              resolver: "Simbad",
              searchConeRadius: "12.1",
              searchConeRadiusUnits: "arcminutes",
              targetTypes: new Set<TargetType>()
            }}
            onChange={onChange}
          />
        )
      )
    ).toMatchSnapshot();
  });

  it("should call onChange with targetName", () => {
    const target = wrapper.find('input[data-test="target-name-input"]');
    target.simulate("change", "");
    expect(onChange).toBeCalled();
  });

  it("should call onChange with right ascension", () => {
    let value: any;
    let event: any;
    const resolverForm = wrapper.find(
      'input[data-test="right-ascension-input"]'
    );
    const resolver = resolverForm.find("input");
    value = "123.1";
    event = { target: { value, name: "rightAscension" } };
    resolver.simulate("change", event);
    expect(onChange).toBeCalledWith({
      errors: { rightAscension: "" },
      resolver: "Simbad",
      rightAscension: "123.1",
      searchConeRadiusUnits: "arcseconds",
      targetTypes: new Set<TargetType>()
    });

    value = "";
    event = { target: { value, name: "rightAscension" } };
    resolver.simulate("change", event);
    expect(onChange).toBeCalledWith({
      errors: { rightAscension: "" },
      resolver: "Simbad",
      rightAscension: "",
      searchConeRadiusUnits: "arcseconds",
      targetTypes: new Set<TargetType>()
    });

    value = undefined;
    event = { target: { value, name: "rightAscension" } };
    resolver.simulate("change", event);
    expect(onChange).toBeCalledWith({
      errors: { rightAscension: "" },
      resolver: "Simbad",
      rightAscension: "",
      searchConeRadiusUnits: "arcseconds",
      targetTypes: new Set<TargetType>()
    });
  });

  it("should call onChange for selectors as well", () => {
    const resolver = wrapper.find('select[data-test="resolver-select"]');
    resolver.simulate("change", "");
    expect(onChange).toBeCalled();

    const rUnits = wrapper.find('select[data-test="radius-units-select"]');
    rUnits.simulate("change", "");
    expect(onChange).toBeCalled();
  });
});
