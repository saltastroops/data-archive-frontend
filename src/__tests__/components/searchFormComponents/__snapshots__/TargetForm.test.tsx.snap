// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetForm should render correctly 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <p>
        Target name
      </p>
      <InputField
        data-test="target-name-input"
        disabled={false}
        name="name"
        onChange={[Function]}
        value=""
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.div>
          <p>
            Resolver
          </p>
          <SelectField
            data-test="resolver-select"
            name="resolver"
            onChange={[Function]}
            value="Simbad"
          >
            <option
              value="Simbad"
            >
              Simbad
            </option>
            <option
              value="NED"
            >
              NED
            </option>
            <option
              value="VizieR"
            >
              VizieR
            </option>
          </SelectField>
        </styled.div>
        <styled.div>
          <br />
          <button
            className="button is-info false"
            disabled={false}
            onClick={[Function]}
            title=""
            type="button"
          >
            resolve
          </button>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <p>
        Right ascension
      </p>
      <InputField
        data-test="right-ascension-input"
        disabled={false}
        name="rightAscension"
        onChange={[Function]}
        value=""
      />
    </styled.div>
    <styled.div>
      <p>
        Declination
      </p>
      <InputField
        data-test="declination-input"
        disabled={false}
        name="declination"
        onChange={[Function]}
        value=""
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <p>
        Search radius
      </p>
      <InputField
        data-test="search-cone-radius-input"
        name="searchConeRadius"
        onChange={[Function]}
        value=""
      />
    </styled.div>
    <styled.div>
      <p>
        Radius units
      </p>
      <SelectField
        data-test="radius-units-select"
        name="radiusUnits"
        onChange={[Function]}
        value="arcseconds"
      >
        <option
          value="arcseconds"
        >
          Arcseconds
        </option>
        <option
          value="arcminutes"
        >
          Arcminutes
        </option>
        <option
          value="degrees"
        >
          Degrees
        </option>
      </SelectField>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <p>
        Target type
      </p>
      <TargetTypesSelector
        onChange={[Function]}
        targetTypes={Set {}}
      />
    </styled.div>
  </styled.div>
</Fragment>
`;

exports[`TargetForm should render correctly 2`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <p>
        Target name
      </p>
      <InputField
        data-test="target-name-input"
        disabled={false}
        name="name"
        onChange={[Function]}
        value="NGC 1234"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.div>
          <p>
            Resolver
          </p>
          <SelectField
            data-test="resolver-select"
            name="resolver"
            onChange={[Function]}
            value="Simbad"
          >
            <option
              value="Simbad"
            >
              Simbad
            </option>
            <option
              value="NED"
            >
              NED
            </option>
            <option
              value="VizieR"
            >
              VizieR
            </option>
          </SelectField>
        </styled.div>
        <styled.div>
          <br />
          <button
            className="button is-info false"
            disabled={false}
            onClick={[Function]}
            title=""
            type="button"
          >
            resolve
          </button>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <p>
        Right ascension
      </p>
      <InputField
        data-test="right-ascension-input"
        disabled={false}
        name="rightAscension"
        onChange={[Function]}
        value=""
      />
    </styled.div>
    <styled.div>
      <p>
        Declination
      </p>
      <InputField
        data-test="declination-input"
        disabled={false}
        name="declination"
        onChange={[Function]}
        value=""
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <p>
        Search radius
      </p>
      <InputField
        data-test="search-cone-radius-input"
        name="searchConeRadius"
        onChange={[Function]}
        value=""
      />
    </styled.div>
    <styled.div>
      <p>
        Radius units
      </p>
      <SelectField
        data-test="radius-units-select"
        name="radiusUnits"
        onChange={[Function]}
        value="arcseconds"
      >
        <option
          value="arcseconds"
        >
          Arcseconds
        </option>
        <option
          value="arcminutes"
        >
          Arcminutes
        </option>
        <option
          value="degrees"
        >
          Degrees
        </option>
      </SelectField>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <p>
        Target type
      </p>
      <TargetTypesSelector
        onChange={[Function]}
        targetTypes={Set {}}
      />
    </styled.div>
  </styled.div>
</Fragment>
`;

exports[`TargetForm should render correctly 3`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <p>
        Target name
      </p>
      <InputField
        data-test="target-name-input"
        disabled={false}
        error="this is an error"
        name="name"
        onChange={[Function]}
        value=""
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.div>
          <p>
            Resolver
          </p>
          <SelectField
            data-test="resolver-select"
            name="resolver"
            onChange={[Function]}
            value="Simbad"
          >
            <option
              value="Simbad"
            >
              Simbad
            </option>
            <option
              value="NED"
            >
              NED
            </option>
            <option
              value="VizieR"
            >
              VizieR
            </option>
          </SelectField>
        </styled.div>
        <styled.div>
          <br />
          <button
            className="button is-info false"
            disabled={false}
            onClick={[Function]}
            title=""
            type="button"
          >
            resolve
          </button>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <p>
        Right ascension
      </p>
      <InputField
        data-test="right-ascension-input"
        disabled={false}
        name="rightAscension"
        onChange={[Function]}
        value=""
      />
    </styled.div>
    <styled.div>
      <p>
        Declination
      </p>
      <InputField
        data-test="declination-input"
        disabled={false}
        name="declination"
        onChange={[Function]}
        value=""
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <p>
        Search radius
      </p>
      <InputField
        data-test="search-cone-radius-input"
        name="searchConeRadius"
        onChange={[Function]}
        value="12.1"
      />
    </styled.div>
    <styled.div>
      <p>
        Radius units
      </p>
      <SelectField
        data-test="radius-units-select"
        name="radiusUnits"
        onChange={[Function]}
        value="arcminutes"
      >
        <option
          value="arcseconds"
        >
          Arcseconds
        </option>
        <option
          value="arcminutes"
        >
          Arcminutes
        </option>
        <option
          value="degrees"
        >
          Degrees
        </option>
      </SelectField>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <p>
        Target type
      </p>
      <TargetTypesSelector
        onChange={[Function]}
        targetTypes={Set {}}
      />
    </styled.div>
  </styled.div>
</Fragment>
`;
